cdap-ui/cypress/integration/pipeline.canvas.actions.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { INodeInfo, INodeIdentifier } from '../typings';
+let headers = {};
+
+describe('Pipeline Canvas actions', () => {
+  // Uses API call to login instead of logging in manually through UI
+  before(() => {
+    loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
+        };
+      });
+    });
+  });
+
+  beforeEach(() => {
+    getArtifactsPoll(headers);
+  });
+
+  it.skip('Should correctly undo/redo actions done by the user', () => {
+    cy.visit('/pipelines/ns/default/studio');
+    const undoSelector = '[data-cy="pipeline-undo-action-btn"]';
+    const redoSelector = '[data-cy="pipeline-redo-action-btn"]';
+    // During the initial load of studio these buttons should be disabled
+    cy.get(redoSelector).should('have.attr', 'disabled', 'disabled');
+    cy.get(undoSelector).should('have.attr', 'disabled', 'disabled');
+
+    cy.create_simple_pipeline();
+    cy.get(undoSelector).should('not.have.attr', 'disabled');
+    cy.get(redoSelector).should('have.attr', 'disabled', 'disabled');
+
+    // Undo the connections
+    cy.get(undoSelector).click();
+    cy.get(undoSelector).click();
+
+    cy.get_pipeline_json().then(pipelineConfig => {
+      const connections = pipelineConfig.config.connections;
+      expect(connections.length).eq(0);
+    });
+
+    // Redo one connection (connection between source & transform)
+    cy.get(redoSelector).click();
+    cy.get_pipeline_json().then(pipelineConfig => {
+      const connections = pipelineConfig.config.connections;
+      expect(connections.length).eq(1);
+    });
+
+    // Undo everything.
+    cy.get(undoSelector).click();
+    cy.get(undoSelector).click();
+    cy.get(undoSelector).click();
+    cy.get(undoSelector).click();
+    // By now it should be an empty canvas and the export button should be disabled.
+    cy.get('[data-cy="pipeline-export-btn"]').should('have.attr', 'disabled', 'disabled');
+
+    // Redo everything.
+    cy.get(redoSelector).click();
+    cy.get(redoSelector).click();
+    cy.get(redoSelector).click();
+    cy.get(redoSelector).click();
+    cy.get(redoSelector).click();
+
+    cy.get('[data-cy="pipeline-export-btn"]').should('not.have.attr', 'disabled');
+    cy.get_pipeline_json().then(pipelineConfig => {
+      const connections = pipelineConfig.config.connections;
+      const stages = pipelineConfig.config.stages;
+      expect(connections.length).eq(2);
+      expect(stages.length).eq(3);
+    });
+  });
+
+  it('Should fit pipeline to screen', () => {
+    cy.visit('/pipelines/ns/default/studio');
+    cy.create_complex_pipeline();
+    cy.get('[data-cy="plugin-node-BigQueryMultiTable-batchsink-7"]').should('be.visible');
+    cy.get('[data-cy="plugin-node-BigQueryMultiTable-batchsink-6"]').should('be.visible');
+    cy.get('[data-cy="pipeline-zoom-in-control"]').click();
```

Comment:
It really helped readability in the previous test to set the selector string to a variable - could do for the other control buttons too.